prow/cmd/sinker/main.go

```diff
@@ -138,13 +138,38 @@ func main() {
 }
 
 type controller struct {
-	logger        *logrus.Entry
-	prowJobClient prowv1.ProwJobInterface
-	podClients    []corev1.PodInterface
-	config        config.Getter
+	logger         *logrus.Entry
+	prowJobClient  prowv1.ProwJobInterface
+	podClients     []corev1.PodInterface
+	config         config.Getter
+	currentMetrics *sinkerReconciliationMetrics
+}
+
+type sinkerReconciliationMetrics struct {
+	podsCreatedByProw      int
+	startAt                time.Time
+	finishedAt             time.Time
+	podsTotalRemovedByProw int
+	podsRemovedByProw      map[string]int
+}
+
+func (c *sinkerReconciliationMetrics) addPodByReason(reason string, logger *logrus.Entry) {
+	if c.podsRemovedByProw == nil {
+		c.podsRemovedByProw = make(map[string]int)
+	}
+	if len(reason)==0 {
+		logger.Warn("Warn len(reason) is 0.")
```

Comment:
`Warn` will stutter in the log, also, maybe a little more informative message would be helpful.

Reply:
The function is removed now.